### PR TITLE
Make sandbox install script WSL-friendly

### PR DESCRIPTION
**Pull request by kispaljr**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
> /kind bug
> /kind cleanup
> /kind design
> /kind documentation
> /kind failing-test

/kind feature

> /kind flake

**What this PR does / why we need it**:

The install_sandbox.sh script currently fails on WSL (Windows Subsystem for Linux). This PR adds support for WSL VMs. 
The introduced changes are:
- use a "local" connection for running the ansible playbook to mitigate the missing SSHD in WSL
- changes in the find command that tries to locate the mitogen python package: the new version avoids delving into mounted, potentially large and slow filesystems (like the mounted windows drives in WSL)
- the script prints the name of the steps it takes to make it easier to follow the progress and also to ease troubleshooting

I only tested this on WSL Ubuntu.

**Which issue(s) this PR fixes**:

Fixes https://github.com/nephio-project/nephio/issues/802

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: e2e/provision/install_sandbox.sh [hunk 1]
```diff
@@ -15,7 +15,13 @@ set -o nounset
 
 export HOME=${HOME:-/home/ubuntu/}
 
+function h1 {
+    echo
+    echo "** $*"
+}
```
[the remark this answers is not in the document]
Comment (kispaljr) [reply e2e/provision/install_sandbox.sh]: I ended up calling it print_task_header(). what do you think?

**Comment by kispaljr** [comment]
I believe the free5gc-ubuntu-focal test is failing because, due to changed timing of parallel process, it hits this race condition issue: https://github.com/nephio-project/nephio/issues/803.

That issue should and will be fixed in the porch repo. I suggest to ignore this faulty test for this PR, since the change in this PR doesn't causes it to fail.

**Review by liamfallon — COMMENTED** [review]
Tested on Ubuntu and these changes don't break current install.

/approve
/lgtm

**Comment by liamfallon** [comment]
New porchctl binary is now available so the tests may pass.

**Comment by kispaljr** [comment]
> New porchctl binary is now available so the tests may pass.

Do the e2e tests actually use the new porchctl binary? It seems that the e2e docker image hasn't been updated for a while: https://hub.docker.com/r/nephio/e2e . I am not familiar with the actual terraform process, so I this is an honest question.

**Comment by efiacor** [comment]
> > New porchctl binary is now available so the tests may pass.
> 
> Do the e2e tests actually use the new porchctl binary? It seems that the e2e docker image hasn't been updated for a while: https://hub.docker.com/r/nephio/e2e . I am not familiar with the actual terraform process, so I this is an honest question.

It should be using the "latest" build of the binary yes.
https://github.com/nephio-project/test-infra/blob/main/e2e/provision/playbooks/cluster.yml#L136
How to verify, not sure.

**Comment by liamfallon** [comment]
I think this PR needs to be rebased to pick up the changes to the test infra.

**Pull request merged**